syntax = "proto3";

package financial;

option java_multiple_files = true;
option java_package = "com.finsight.proto";
option java_outer_classname = "FinancialDataProto";

service FinancialDataService {

  rpc GetStockPrice (StockRequest) returns (StockPriceResponse);
  
  rpc GetBatchPrices (BatchStockRequest) returns (BatchStockResponse);
}

message StockRequest {
  string symbol = 1;
}

message StockPriceResponse {
  string symbol = 1;
  double price = 2;
  string timestamp = 3; // ISO 8601 string time
}

message BatchStockRequest {
  repeated string symbols = 1;
}

message BatchStockResponse {
  repeated StockPriceResponse prices = 1;
  int32 success_count = 2;
  int32 failed_count = 3;
}